pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java

```diff
@@ -69,23 +70,55 @@ public void stop() {
     depositProcessingController.stopIfSubscribed();
   }
 
-  private SafeFuture<Void> headBeforeMinGenesisMode(EthBlock.Block headBlock) {
+  private SafeFuture<Void> processStart(
+      final EthBlock.Block headBlock, final ReplayDepositsResult replayDepositsResult) {
+    BigInteger startBlockNumber =
+        replayDepositsResult.getBlockNumber().orElse(UnsignedLong.ZERO).bigIntegerValue();
```

Comment:
We've got this logic in at least three places - probably worth either extracting a method for it or calculating up-front and passing the value down.